app/queries/customers_with_balance.rb

```diff
@@ -1,26 +1,41 @@
 # frozen_string_literal: true
 
-# Fetches the customers of the specified enterprise including the aggregated balance across the
-# customer's orders. That is, we get the total balance for each customer with this enterprise.
+# Adds an aggregated 'balance_value' to each customer based on their order history
+#
 class CustomersWithBalance
-  def initialize(enterprise)
+  def initialize(
+    enterprise: nil,  # Filter customers by enterprise
+    customers: nil    # Filter customers by record/collection of customers/ids
+  )
     @enterprise = enterprise
+    @customers = customers
+
+    validate_arguments
```

Comment:
We can simplify this by just taking a customers relation in the initializer. This module wouldn't need to know `enterprise` at all. Other places can then call it with `Customer.of(enterprise)`. Much better division of responsibilities.

Reply:
Fixed